### PR TITLE
Add redis-streams backend

### PR DESCRIPTION
**Pull request by alex-oleshkevich**
Based on https://github.com/encode/broadcaster/pull/102
Seems that the author of https://github.com/encode/broadcaster/pull/102 did not notice a call for update, so I am creating a new PR in order to proceed with this backend.

@encode/maintainers please do a review

closes https://github.com/encode/broadcaster/pull/102
closes https://github.com/encode/broadcaster/pull/3

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -111,6 +112,6 @@ state, make sure to strictly pin your requirements to `broadcaster==0.2.0`.
 To be more capable we'd really want to add some additional backends, provide API support for reading recent event history from persistent stores, and provide a serialization/deserialization API...
 
 * Serialization / deserialization to support broadcasting structured data.
-* Backends for Redis Streams, Apache Kafka, and RabbitMQ.
+* Backends for Apache Kafka, and RabbitMQ.
```
Comment (tomchristie) [inline README.md:115]: 🤔

Comment (alex-oleshkevich) [reply README.md]: This is `What's next` section. Redis Streams lands with this PR.

**Comment by tomchristie** [comment]
Are you able to give a brief pointer towards when there'd be a use-case for redis streams vs. redis pub/sub?

**Comment by alex-oleshkevich** [comment]
Redis Streams is a lightweight Kafka alternative. While pub sub is fire-and-forget, Redis stream is an append only log.

**Comment by tomchristie** [comment]
Reviews README and discovers that past me already wrote about this. Thanks that person.

> To be more capable we'd really want to [...]
>
> Add support for `subscribe('chatroom', history=100)` for backends which provide persistence. (Redis Streams, Apache Kafka) This will allow applications to subscribe to channel updates, while also being given an initial window onto the most recent events. We *might* also want to support some basic paging operations, to allow applications to scan back in the event history.

**Comment by tsotnesharvadze** [comment]
thanks 🙏

**Pull request merged**